apps/src/blockly/addons/cdoUtils.js

```diff
@@ -109,7 +109,39 @@ export function getCode(workspace) {
   // return JSON.stringify(Blockly.serialization.workspaces.save(workspace));
 }
 
-// TODO: Re-define with a new custom field.
-export function soundField(onChange) {
-  return new Blockly.FieldDropdown([['Choose', 'Choose']], onChange);
+export function soundField(onChange, onDisplay, icon) {
+  // Handle 'play sound' block with default param from CDO blockly.
+  // TODO: Remove when sprite lab is migrated to Google blockly.
+  const validator = newValue => {
+    if (typeof newValue !== 'string') {
+      return null;
+    }
+    if (newValue === 'Choose') {
+      return DEFAULT_SOUND;
+    }
+    return newValue;
+  };
+  // FieldButton(value, validator, onChange, onDisplay, icon)
+  return new Blockly.FieldButton(
+    DEFAULT_SOUND,
+    validator,
+    onChange,
+    onDisplay,
+    icon
+  );
+}
+
+export function locationField(icon, onChange) {
+  const onDisplaySetField = value => {
+    if (value) {
+      try {
+        const loc = JSON.parse(value);
+        return `(${loc.x}, ${APP_HEIGHT - loc.y})`;
+      } catch (e) {
+        // Just ignore bad values
+      }
+    }
+  };
+  // FieldButton(value, validator, onChange, onDisplay, icon)
```

Comment:
nit: remove? edit: oh, realizing this might be here for documentation purposes. If we're concerned that these arguments may seem ambiguous, one option could be passing object into the constructor of cdoFieldButton instead of separate args? i.e. `FieldButton({onChange, onDisplay: onDisplaySetField, icon})`. It would also be good to clearly indicate that `value` and `validator` are optional fields in the documentation for cdoFieldButton's constructor

Reply:
Ooo - I like that! And it simplifies the `fromJson` as well - thanks!